src/pages/settings/Subscription/CardSection/utils.ts

```diff
@@ -139,5 +151,23 @@ function getNextBillingDate(): string {
     return format(nextBillingDate, CONST.DATE.MONTH_DAY_YEAR_FORMAT);
 }
 
+function getPurchaseDetails(purchase?: Purchase) {
+    const purchaseAmount = purchase?.message.billableAmount;
+    const purchaseCurrency = purchase?.currency;
+    const isBillingTypeProper = purchase?.message.billingType === 'failed_2018';
+    const purchaseCurrencySymbol = getLocalizedCurrencySymbol(purchaseCurrency ?? CONST.CURRENCY.USD);
+    const purchaseDate = purchase?.created;
+    const purchaseDateFormatted = purchaseDate ? DateUtils.formatWithUTCTimeZone(purchaseDate, CONST.DATE.MONTH_DAY_YEAR_FORMAT) : null;
+
+    const formattedAmount = convertToFrontendAmountAsString(purchaseAmount, purchaseCurrency);
+
+    return {
+        purchaseAmountWithCurrency: purchaseCurrencySymbol + formattedAmount,
+        isBillingTypeProper,
+        purchaseDateFormatted,
+        isPurchase: !!purchase,
+    };
+}
+
```

Comment:
Can we remove this function? It does not much but return fields from the passed object and the functiion is used once.

Also regarding the formatted amount with currency let's use `convertAmountToDisplayString` / `convertToShortDisplayString`